### PR TITLE
MAINT: Correct and update polynomial package documentation.

### PR DESCRIPTION
**Pull request by jeffyancey**
While surveying the polynomial package I attempted to update `Examples`, `See Also`, and make other miscellaneous changes I thought necessary.

I I believe all my changes are in compliance with the official documentation policy, but any feedback is very much appreciated.

### CODE REVIEW

**Comment by jeffyancey** [comment]
Looks like I didn't push a couple of my reversions. I will do so and respond tomorrow. Thanks for the feedback @eric-wieser

File: numpy/polynomial/_polybase.py [hunk 1]
```diff
@@ -980,7 +972,7 @@ def basis(cls, deg, domain=None, window=None):
             Degree of the basis polynomial for the series. Must be >= 0.
         domain : {None, array_like}, optional
             If given, the array must be of the form ``[beg, end]``, where
-            ``beg`` and ``end`` are the endpoints of the domain. If None is
+            `beg` and `end` are the endpoints of the domain. If None is
```
[the remark this answers is not in the document]
Comment (mattip) [reply numpy/polynomial/_polybase.py]: Just to clarify: in sphinx, as opposed to markdown, text inside a single backtick is a [cross-reference](http://www.sphinx-doc.org/en/stable/tutorial.html#documenting-objects) but without a [domain](http://www.sphinx-doc.org/en/stable/domains.html),  meant to turn into a link using the default domain role. Text inside a double-backtick is formatted as code.

File: numpy/polynomial/chebyshev.py [hunk 2]
```diff
@@ -1192,32 +1212,33 @@ def chebval2d(x, y, c):
...
-        unchanged and if it isn't an ndarray it is treated as a scalar.
+        The 2-D series is evaluated at the points `(x, y)`, where `x` and `y`
+        must have the same shape. `x` or `y` are converted to ndarrays if
+        they are tuples or lists, otherwise they are left unchanged. If either
+        series isn't an ndarray, tuple, or list they are treated as scalars and
+        must have the same shape after conversion.
     c : array_like
         Array of coefficients ordered so that the coefficient of the term
-        of multi-degree i,j is contained in ``c[i,j]``. If `c` has
+        of multi-degree i,j is contained in c[i,j]. If `c` has
```
[the remark this answers is not in the document]
Comment (jeffyancey) [reply numpy/polynomial/chebyshev.py]: I agree. Will revert.

File: numpy/polynomial/legendre.py [hunk 3]
```diff
@@ -1513,6 +1551,13 @@ def legfit(x, y, deg, rcond=None, full=False, w=None):
 
     Examples
     --------
+    >>> from numpy.polynomial import legendre as L
+    >>> c = (1,2,3)
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply numpy/polynomial/legendre.py]: Ah, that's a good reason to leave it like this. Let's aim for consistency first, and then in another CR we can always overhaul every single example

**Comment by jeffyancey** [comment]
Reversions have gone through.

File: numpy/polynomial/legendre.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeffyancey) [reply numpy/polynomial/legendre.py]: I'd like to arrive at a consensus on which of the two example approaches I've seen throughout this directory is preferable:

A:
```
>>> from numpy.polynomial.hermite_e import hermemulx
>>> hermemulx([1, 2, 3])
array([ 2.,  7.,  2.,  3.])
```
B:
```
>>> from numpy.polynomial import chebyshev as C
>>> c = (1,2,3)
>>> C.chebmulx(c)
array([ 1. ,  2.5,  1. ,  1.5])
```

File: numpy/polynomial/chebyshev.py [hunk 4]
```diff
@@ -1156,6 +1167,15 @@ def chebval(x, c, tensor=True):
...
     --------
+    >>> from numpy.polynomial import chebyshev as C
+    >>> coef = (1,2,3)
+    >>> C.chebval(1, coef)
+    6.0
+    >>> c1 = (1,2)
+    >>> c2 = (3,4)
+    >>> C.chebval((c1, c2), coef)
+    array([[   6.,   26.],
+           [  58.,  102.]])
```
[the remark this answers is not in the document]
Comment (jeffyancey) [reply numpy/polynomial/chebyshev.py]: I am on version 1.15.0. I'll doublecheck that is the only version I have installed and that jupyter is 1) using it and 2) not doing anything funny to my output. Thanks for alerting me to this.

File: numpy/polynomial/legendre.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeffyancey) [reply numpy/polynomial/legendre.py]: Once we have reached a consensus, I will make a second PR to go through and have all examples conform, if it is desirable.

**Comment by jeffyancey** [comment]
looks like I added all the newly merged commits to my PR accidentally. Sorry all. My changes are  limited to `numpy/polynomial` files.

**Comment by charris** [comment]
Might try a rebase on master. If all else fails, google `git reflog` which will allow you to reset git to a previous state.

**Comment by jeffyancey** [comment]
whoops, slipped my mind to come back and fix this. Will be done shortly.

**Comment by jeffyancey** [comment]
closing and repopening in a clean PR.

**Pull request closed without merging**